UPGRADE_GUIDE.md

```diff
@@ -2,9 +2,146 @@
 
 Use the following guide to assist in the upgrade process of the `easypost-csharp` library between major versions.
 
+- [Upgrading from 4.x to 5.0](#upgrading-from-4x-to-50)
 - [Upgrading from 3.x to 4.x](#upgrading-from-3x-to-40)
 - [Upgrading from 2.x to 3.0](#upgrading-from-2x-to-30)
 
+## Upgrading from 4.x to 5.0
+
+### 5.0 High Impact Changes
+
+- [Service Functions](#50-service-functions)
+- [Client Configuration](#50-client-configuration)
+
+### 5.0 Medium Impact Changes
+
+- [Naming Conventions](#50-naming-conventions)
+- [Exception Changes](#50-exception-changes)
+- [Cancellation Tokens](#50-cancellation-tokens)
+- [Parameter Sets](#50-parameter-sets)
+
+### 5.0 Low Impact Changes
+
+- [Drop RestSharp Dependency](#50-drop-restsharp-dependency)
+- [Dispose Pattern](#50-dispose-pattern)
+- [Docstring Improvements](#50-docstring-improvements)
+
+## 5.0 Service Functions
+
+*Likelihood of Impact: **High***
+
+Following the design for the other EasyPost SDKs, all functions have been consolidated into appropriate service classes.
+
+Previously, only `Create`, `Retrieve` and `All` functions were available in services, with all other functions related to a specific instance of a resource available on the resource itself.
+
+For example, v4.x flow of creating a pickup and then buying it:
+
+```csharp
+var pickup = await myClient.Pickup.Create(parameters);
+pickup.Buy();  // pickup variable is updated in-place
+```
+
+In v5.0, the flow is now:
+
+```csharp
+var pickup = await myClient.Pickup.Create(parameters);
+pickup = await myClient.Pickup.Buy(pickup.Id); // re-assign pickup variable to capture the updated Pickup object
```

Comment:
suggestion: to avoid confusion, maybe consider calling the variable `boughtPickup`. Users will know they can re-assign vars but that will help distinguish there is a change here.